fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/cursors/MapPipelinedCursor.java

```diff
@@ -90,15 +93,17 @@ public CompletableFuture<RecordCursorResult<V>> onNext() {
 
     @Override
     public void close() {
-        while (!pipeline.isEmpty()) {
```

Comment:
Should we do the same thing to the `FlatMapPipelinedCursor`, which I believe can hit similar issues?

Reply:
I filed a new issue about this: #2953